cfme/fixtures/provider.py

```diff
@@ -382,3 +386,38 @@ def console_template(provider):
 @pytest.fixture(scope="module")
 def console_template_modscope(provider):
     return _get_template(provider, 'console_template')
+
+
+@pytest.mark.hookwrapper
+def pytest_fixture_setup(fixturedef, request):
+    # since we use DataProvider at collection time and BaseProvider in fixtures and tests,
+    # we need to instantiate BaseProvider and replace DataProvider obj with it right before first
+    # provider fixture request.
+    # There were several other ways to do that. However, those bumped into different
+    # scope mismatch issues.
+    if fixturedef.argname == 'provider':
+        kwargs = {}
+        for argname in fixturedef.argnames:
+            fixdef = request._get_active_fixturedef(argname)
+            result, arg_cache_key, exc = fixdef.cached_result
+            request._check_scope(argname, request.scope, fixdef.scope)
+            kwargs[argname] = result
+
+        fixturefunc = fixturedef.func
+        if request.instance is not None:
+            fixturefunc = getimfunc(fixturedef.func)
+            if fixturefunc != fixturedef.func:
+                fixturefunc = fixturefunc.__get__(request.instance)
```

Comment:
**QUESTION**
Do we really need to use `__get__()`? Why not `getattr(fixturefunc, request.instance)`

Reply:
I found this part of code somewhere in pytest while I was looking for a way to override fixture_setup for provider fixture. I suspect there is a reason to use __get__ descriptor. I can figure out exact reason if you want.

Reply:
this is to invoke a method descriptor if necessary - a fixture declared in a class needs to be bound to the correct instance before invoking